Sources/TuistSupport/Utils/GitEnvironment.swift

```diff
@@ -0,0 +1,67 @@
+import Combine
+import CombineExt
+import Foundation
+import TSCUtility
+
+public protocol GitEnvironmenting {
+    /// It treturns the environment's Git credentials for GitHub.
+    /// This is useful for operations such pulling newer Tuist versions from GitHub
+    /// without hitting API limits.
+    func githubCredentials() -> Deferred<Future<(username: String, password: String)?, Error>>
```

Comment:
Instead of using tuple, I'd probably create a new model `struct GithubCredentials`

Reply:
Good idea! I updated it to use a struct instead.